src/img/node_modules/docker-registry-client/lib/registry-client-v2.js

```diff
@@ -1590,6 +1593,127 @@ RegistryClientV2.prototype.createBlobReadStream =
+                    } else {
+                        accept = [self._headers.accept];
+                    }
+                }
+                accept.push(MEDIATYPE_MANIFEST_V2);
+                if (acceptManifestLists) {
+                    accept.push(MEDIATYPE_MANIFEST_LIST_V2);
+                }
+                headers = common.objMerge({}, self._headers, {accept: accept});
+            }
+            self._api.get({
+                path: fmt('/v2/%s/manifests/%s',
+                    encodeURI(self.repo.remoteName),
+                    encodeURI(opts.ref)),
+                headers: headers
+            }, function _afterCall(err, req, res_, manifest_, body) {
+                if (err) {
+                    return next(err);
+                }
+
+                manifest = manifest_;
+
+                if (manifest.schemaVersion === 1) {
+                    try {
+                        var jws = _jwsFromManifest(manifest, body);
+                        // Some v2 registries (Amazon ECR) do not provide the
+                        // 'docker-content-digest' header.
+                        if (res_.headers['docker-content-digest']) {
+                            _verifyManifestDockerContentDigest(res_, jws);
+                        } else {
+                            self.log.debug({headers: res_.headers},
+                                'no Docker-Content-Digest header on ' +
+                                'getManifest response');
+                        }
+                        _verifyJws(jws);
+                    } catch (verifyErr) {
+                        return next(verifyErr);
+                    }
+                }
+
+                if (manifest.schemaVersion > maxSchemaVersion) {
+                    cb(new restifyErrors.InvalidContentError(fmt(
+                        'unsupported schema version %s in %s:%s manifest',
+                        manifest.schemaVersion, self.repo.localName,
+                        opts.ref)));
+                    return;
+                }
+
+                // Get the manifest digest for the correct platform
+                if (manifest.mediaType === MEDIATYPE_MANIFEST_LIST_V2
+                     || manifest.mediaType ===
+                     MEDIATYPE_OCI_MANIFEST_LIST_V1 ) {
+                    if (!Array.isArray(manifest.manifests) ||
+                            manifest.manifests.length === 0) {
+                        cb(new restifyErrors.InvalidContentError(fmt(
+                            'no manifests in %s:%s manifest list',
+                            self.repo.localName, opts.ref)));
+                        return;
+                    }
+                    for (var i = 0; i < manifest.manifests.length;  i++) {
+                        var platform = manifest.manifests[i].platform;
+                        if (platform &&
+                            platform.architecture === 'amd64' &&
```

Comment:
I honestly cannot remember if we ONLY ever want amd64 of if i386 is okay as well?  I'm happy to be educated here.